src/main/java/net/neoforged/neoforge/common/extensions/IBlockExtension.java

```diff
@@ -112,6 +112,40 @@ default float getFriction(BlockState state, LevelReader level, BlockPos pos, @Nu
         return self().getFriction();
     }
 
+    /**
+     * Gets the speed factor at the given location at the given state. Normally
+     * between 0 and 1.
+     * <p>
+     * Note that entities may manipulate the speed factor on their own;
+     * for {@link LivingEntity#getBlockSpeedFactor()}
+     *
+     * @param state  the state of the block
+     * @param level  the level
+     * @param pos    the position in the level
+     * @param entity the entity in question
+     * @return the factor by which the entity's speed should be multiplied
+     */
+    default float getSpeedFactor(BlockState state, LevelReader level, BlockPos pos, @Nullable Entity entity) {
+        return self().getSpeedFactor();
+    }
+
+    /**
+     * Gets the jump factor at the given location at the given state. Normally
+     * between 0 and 1.
```

Comment:
The way I would explain this is that it returns the scalar applied to the movement vector in the Y direction. This value is then multiplied with the jump strength attribute to obtain the final upward momentum. Two more notes I would add is if the factor is greater than or equal to 1, then the entity can auto jump off this block when the setting is enabled. The other would be if the jump factor is too small (0 or less in most cases), then the entity will not jump at all.